bedshift/bedshift.py

```diff
@@ -247,16 +246,24 @@ def shift_from_file(self, fp, shiftrate, shiftmean, shiftstdev, delimiter="\t"):
         num_shift = int(rows * shiftrate)
 
         intersect_regions = self._find_overlap(fp)
-        interlen = len(intersect_regions)
+        rows2shift = intersect_regions.sample(frac=shiftrate)
+        indices2shift = []
+        for _, r in rows2shift.iterrows():
```

Comment:
Here is a method that is probably more performant: 

<img width="369" alt="image" src="https://user-images.githubusercontent.com/32209687/112402435-878f3280-8cda-11eb-926d-aaa632f59c7e.png">


Basically to find the common rows, you perform a merge. By using `reset_index` the merge will keep the indices from `self.bed` in the output, which you can then pass to `self.shift`